fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java

```diff
@@ -1250,4 +1266,110 @@ private static TFetchSchemaTableDataResult tablePropertiesMetadataResult(TSchema
         result.setStatus(new TStatus(TStatusCode.OK));
         return result;
     }
+
+    private static void partitionsForInternalCatalog(UserIdentity currentUserIdentity,
+                CatalogIf catalog, DatabaseIf database, List<TableIf> tables, List<TRow> dataBatch) {
+        for (TableIf table : tables) {
+            if (!(table instanceof OlapTable)) {
+                continue;
+            }
+            if (!Env.getCurrentEnv().getAccessManager().checkTblPriv(currentUserIdentity, catalog.getName(),
+                    database.getFullName(), table.getName(), PrivPredicate.SHOW)) {
+                continue;
+            }
+
+            OlapTable olapTable = (OlapTable) table;
+            Collection<Partition> allPartitions = olapTable.getAllPartitions();
+
+            for (Partition partition : allPartitions) {
+                TRow trow = new TRow();
+                trow.addToColumnValue(new TCell().setStringVal(catalog.getName())); // TABLE_CATALOG
+                trow.addToColumnValue(new TCell().setStringVal(database.getFullName())); // TABLE_SCHEMA
+                trow.addToColumnValue(new TCell().setStringVal(table.getName())); // TABLE_NAME
+                trow.addToColumnValue(new TCell().setStringVal(partition.getName())); // PARTITION_NAME
+                trow.addToColumnValue(new TCell().setStringVal("")); // SUBPARTITION_NAME (not available)
+
+                trow.addToColumnValue(new TCell().setIntVal(0)); //PARTITION_ORDINAL_POSITION (not available)
+                trow.addToColumnValue(new TCell().setIntVal(0)); //SUBPARTITION_ORDINAL_POSITION (not available)
+                trow.addToColumnValue(new TCell().setStringVal(
+                        olapTable.getPartitionInfo().getType().toString())); // PARTITION_METHOD
+                trow.addToColumnValue(new TCell().setStringVal("")); // SUBPARTITION_METHOD
+                ArrayList<Expr> expr = olapTable.getPartitionInfo().getPartitionExprs();
+                String partExpr = (expr != null) ? olapTable.getPartitionInfo().getPartitionExprs().toString() : "";
+
+                trow.addToColumnValue(new TCell().setStringVal(partExpr)); // PARTITION_EXPRESSION (not available)
+                trow.addToColumnValue(new TCell().setStringVal("")); // SUBPARTITION_EXPRESSION (not available)
+                trow.addToColumnValue(new TCell().setStringVal(
+                        olapTable.getPartitionInfo().getPartitionRangeString(partition.getId())));
```

Comment:
For partition range, how about refer to mysql?
![image](https://github.com/user-attachments/assets/99910860-4956-4df4-a6a9-49349412a8e4)

Reply:
![image](https://github.com/user-attachments/assets/2d42f65c-1d05-4314-8a8d-87a7d6ca89eb)

Reply:
description is similar to mysql but since we support multiple partition column, still will come as a expression in the patition description. eg:  [('4'), ('6'))